### PR TITLE
QuestManagerTest

### PR DESCRIPTION
**Pull request by jaadams5**
I was looking for "how to" examples and I noticed some of the code in QuestManagerTest was not actually being used.  I annotated three things I thought were intended as tests.  One passed.  Two failed.  I marked the failing tests as disabled.  It is possible that implicitly making the code unused by not annotating as a Test was deliberate but I think annotating and disabling the test is clearer.

@Veracity0 ?

### CODE REVIEW

**Comment by Veracity0** [comment]
Not annotating the tests was a bug.
If they fail when annotated, either the test or the code it was testing should be fixed.

**Comment by Veracity0** [comment]
Try this.
```
diff --git a/src/net/sourceforge/kolmafia/session/QuestManager.java b/src/net/sourceforge/kolmafia/session/QuestManager.java
index 11710ce0da..6bbed725ef 100644
--- a/src/net/sourceforge/kolmafia/session/QuestManager.java
+++ b/src/net/sourceforge/kolmafia/session/QuestManager.java
@@ -1296,6 +1296,8 @@ public class QuestManager {
     else if (location.contains("action=grandpastory")) {
       if (responseText.contains("bet those lousy Mer-kin up and kidnapped her")) {
         QuestDatabase.setQuestIfBetter(Quest.SEA_MONKEES, "step6");
+      } else if (responseText.contains("that note's definitely Grandma Sea Monkee's handwriting")) {
+        QuestDatabase.setQuestIfBetter(Quest.SEA_MONKEES, "step7");
       } else if (responseText.contains("Gonna need one of them seahorses")) {
         Preferences.setBoolean("corralUnlocked", true);
       }
diff --git a/test/net/sourceforge/kolmafia/session/QuestManagerTest.java b/test/net/sourceforge/kolmafia/session/QuestManagerTest.java
index 5443bfb91b..ba12ddfe3d 100644
--- a/test/net/sourceforge/kolmafia/session/QuestManagerTest.java
+++ b/test/net/sourceforge/kolmafia/session/QuestManagerTest.java
@@ -2100,6 +2100,7 @@ public class QuestManagerTest {
       private static final AdventureResult CHARTREUSE_YARN = ItemPool.get(ItemPool.CHARTREUSE_YARN);
       private static final AdventureResult GRANDMAS_MAP = ItemPool.get(ItemPool.GRANDMAS_MAP);
 
+      @Test
       void talkingToGrandpaStartsQuest() {
         var builder = new FakeHttpClientBuilder();
         var cleanups =
@@ -2172,6 +2173,7 @@ public class QuestManagerTest {
         }
       }
 
+      @Test
       void talkingToGrandpaWithNoteConfirmsQuest() {
         var builder = new FakeHttpClientBuilder();
         var cleanups =
@@ -2196,6 +2198,7 @@ public class QuestManagerTest {
         }
       }
 
+      @Test
       void talkingToGrandpaWithNoteAndYarnAdvancesQuest() {
         var builder = new FakeHttpClientBuilder();
         var cleanups =
@@ -2207,7 +2210,6 @@ public class QuestManagerTest {
                 withItem(CHARTREUSE_YARN));
         try (cleanups) {
           builder.client.addResponse(200, html("request/test_quest_sea_monkee_step_8.html"));
-          builder.client.addResponse(200, ""); // api.php
 
           String URL = "monkeycastle.php?action=grandpastory&topic=note";
           var request = new GenericRequest(URL);
@@ -2220,9 +2222,8 @@ public class QuestManagerTest {
           assertTrue(InventoryManager.hasItem(GRANDMAS_MAP));
 
           var requests = builder.client.getRequests();
-          assertThat(requests, hasSize(2));
+          assertThat(requests, hasSize(1));
           assertPostRequest(requests.get(0), "/monkeycastle.php", "action=grandpastory&topic=note");
-          assertPostRequest(requests.get(1), "/api.php", "what=status&for=KoLmafia");
         }
       }
```

**Comment by jaadams5** [comment]
> Try this.

Done (if I didn't miss anything).

**Comment by Veracity0** [comment]
For future reference:
```
import static internal.helpers.Networking.printRequests;
...
        var requests = client.getRequests();
        printRequests(requests);
        assertThat(requests, hasSize(5));
```
is how you can debug exactly which requests were submitted.
If you were expecting more - or different -- requests, seeing what you actually got is important.

**Comment by Veracity0** [comment]
>  Done (if I didn't miss anything).

You missed one thing, but it's not a big deal.

The test that originally thought it was getting two requests set up two responses.
Since it only got one request (as you can see from the html, which did not request a charpane refresh), the full change would be:

- remove the second "fake response"
- change hasSize(2) to hasSize(1)
- Remove the assert that referenced requests.get(1)

You did the last two, but left the (never used) fake response.

File: test/net/sourceforge/kolmafia/session/QuestManagerTest.java
Comment (Veracity0) [inline test/net/sourceforge/kolmafia/session/QuestManagerTest.java:2225]: As I mentioned, since you are not expecting a second response, you can remove the ```builder.client.addResponse(200, ""); // api.php```

It's not harmful to leave it, so I'm approving, but if you remove it, don't wait for another approval.

**Review by Veracity0 — APPROVED** [review]

File: test/net/sourceforge/kolmafia/session/QuestManagerTest.java
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/session/QuestManagerTest.java:2225]: Removed.  I'm in no hurry...

**Review by Veracity0 — APPROVED** [review]

**Pull request merged**
